backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  forwardRef,
+  HttpException,
+  HttpStatus,
+  Inject,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { CreateUserDto } from '../user/dto/create-user.dto';
+import { UserService } from '../user/user.service';
+import * as bcrypt from 'bcryptjs';
+import * as Response from '../response.messages';
+import { mailer } from '../nodemailer';
+import * as dotenv from 'dotenv';
+import * as env from 'env-var';
+import { sendSms } from '../sms';
+import { User } from '../user/user.model';
+import { JwtDB } from './jwt.model';
+import { InjectModel } from '@nestjs/sequelize';
+import { v4 as uuidv4 } from 'uuid';
+import { GenerateToken } from './dto/generate-token.dto';
+dotenv.config();
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectModel(JwtDB) private jwtRepository: typeof JwtDB,
+    @Inject(forwardRef(() => UserService))
+    private userService: UserService,
+    private jwtService: JwtService,
+  ) {}
+
+  async login(userDto: CreateUserDto): Promise<{ token: string }> {
+    const uuid = uuidv4();
+    const user = await this.validateUser(userDto);
+    const token = await this.generateToken({ ...user, tokenId: uuid });
+    await this.jwtRepository.create({
+      userId: user.id,
+      jwt: token,
+      isActive: true,
+      tokenId: uuid,
+    });
+
+    return { token: token };
+  }
+
+  async logout(userDto: CreateUserDto): Promise<ResponseMsg> {
+    let user: User | null = null;
+    if (userDto.phone)
+      user = await this.userService.getUserByPhone(userDto.phone);
+    else if (userDto.email)
+      user = await this.userService.getUserByEmail(userDto.email);
+    if (!user)
+      throw new HttpException(Response.USER_NOT_FOUND, HttpStatus.NOT_FOUND);
+
+    try {
+      await this.jwtRepository.update(
+        { isActive: false },
+        { where: { userId: user.id } },
+      );
+      return Response.LOG_OUT_SUCESS;
+    } catch {
+      return Response.LOG_OUT_FAILED;
+    }
+  }
+
+  async registration(userDto: CreateUserDto): Promise<string> {
+    const ENCODING_SALT = env.get('ENCODING_SALT').required().asIntPositive();
+    const hashPassword = await bcrypt.hash(userDto.password, ENCODING_SALT);
+
+    const dto = { ...userDto };
+    dto.password = hashPassword;
```

Comment:
```suggestion
    const dto = { ...userDto,  password: hashPassword };
```
не изменяй почем зря